dev-requirements.txt

```diff
@@ -4,13 +4,17 @@ unittest-xml-reporting
 parameterized
 packaging
 transformers
+hypothesis # Avoid test derandomization warning
+sentencepiece # for gpt-fast tokenizer
+expecttest
 
 # For prototype features and benchmarks
 bitsandbytes #needed for testing triton quant / dequant ops for 8-bit optimizers
 matplotlib
 pandas
 fire  # QOL for commandline scripts
 tabulate  # QOL for printing tables to stdout
+numpy # mx benchmarks leverage benchmark timer which uses numpy
```

Comment:
Is there another timer that could be used? 
That being said I can't remember if numpy is a dependency of torch

Reply:
numpy is not supposed to be a dependency of torch but I've seen that assumption broken multiple times - I remember chatting about this with @malfet at some point too

That said the reason why this is needed is because of https://github.com/pytorch/pytorch/pull/128759 

And yes we could just use another timer and drop this dependency completely even from our dev dependencies - although I really need to be writing docs tomorrow so would rather do this in a future PR unless @vkuzo gets to it first

Reply:
Honestly I could also just remove the numpy dependency from dev dependencies because it's only used for a benchmark for mx that doesn't run in CI